### PR TITLE
soc: nxp: imxrt118x: update MPU configuration

### LINKED ISSUE #80721: tests: mimxrt1180_evk: mpu fault cause CI broken
**Describe the bug**

meet MPU fault in mimxrt1180_evk, and seeing Flash error in CI

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 - mimxrt1180_evk
 - What have you tried to diagnose or workaround this issue?
 - git bisect
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?
 - https://github.com/zephyrproject-rtos/zephyr/pull/77600
-->

**To Reproduce**
run below withc 
`
west twister -p mimxrt1180_evk -T samples/hello_world --device-testing --device-serial /dev/ttyACM0
`

**Expected behavior**

test can pass

**Impact**

platform stabilities

**Logs and console output**

when this happen all application will have no log in condole only show Flash error

**Environment (please complete the following information):**

 - OS: (e.g. Linux, )
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used: 193eeaef0c04570b80a954b58ccd98075150fb11

### PR DESCRIPTION
**Pull request by nxp-shelley**
fixes: #80721
Updated mpu region address to secure address.

### CODE REVIEW

**Comment by DerekSnell** [comment]
@hakehuang , can you run our tests on this board, and ensure they are pass with this PR?

@lucien-nxp , can you please review this PR?

File: soc/nxp/imxrt/imxrt118x/m33/mpu_regions.c [hunk 1]
```diff
@@ -7,13 +7,13 @@
 #include <zephyr/devicetree.h>
 #include <zephyr/arch/arm/cortex_m/arm_mpu_mem_cfg.h>
 
-#define REGION_HYPERRAM_BASE_ADDRESS   0x04000000
-#define REGION_HYPERRAM_SIZE           0x04000000
-#define REGION_DTCM_BASE_ADDRESS       0x20000000
+#define REGION_HYPERRAM_BASE_ADDRESS   0x14000000
```
Comment (DerekSnell) [inline soc/nxp/imxrt/imxrt118x/m33/mpu_regions.c:10]:
```suggestion
#define REGION_FLEXSPI2_BASE_ADDRESS   0x14000000
```
I realize you did not name these regions.  But we should not use "HYPERRAM" in the name of these regions, and should call "FLEXSPI2" instead.  This is an SOC file, to use on any board with this SOC.  Our EVK connects HyperRAM to FlexSPI2.  But other boards could use different memory options on FlexSPI2.

File: soc/nxp/imxrt/imxrt118x/m33/mpu_regions.c [hunk 2]
```diff
@@ -7,13 +7,13 @@
...
+#define REGION_HYPERRAM_SIZE           0x00800000
```
Comment (DerekSnell) [inline soc/nxp/imxrt/imxrt118x/m33/mpu_regions.c:11]:
```suggestion
#define REGION_HYPERRAM_SIZE           0x04000000
```
Why did you reduce the FlexSPI2 region size?  The FlexSPI2 memory map is from 0x1400_0000 to 0x17FF_FFFF.  Larger memories than the HyperRAM on our EVK could be connected to FlexSPI2.

File: soc/nxp/imxrt/imxrt118x/m33/mpu_regions.c [hunk 3]
```diff
@@ -7,13 +7,13 @@
...
+#define REGION_DTCM_BASE_ADDRESS       0x30000000
 #define REGION_DTCM_SIZE               0x00020000
-#define REGION_FLEXSPI_BASE_ADDRESS    0x28000000
-#define REGION_FLEXSPI_SIZE            0x08000000
-#define REGION_PERIPHERAL_BASE_ADDRESS 0x40000000
+#define REGION_FLEXSPI_BASE_ADDRESS    0x38000000
+#define REGION_FLEXSPI_SIZE            0x01000000
```
Comment (DerekSnell) [inline soc/nxp/imxrt/imxrt118x/m33/mpu_regions.c:15]:
```suggestion
#define REGION_FLEXSPI_SIZE            0x08000000
```
Same as FlexSPI2 size above, FlexSPI memory map is from 0x3800_0000 to 0x3FFF_FFFF.

**Review by DerekSnell — CHANGES_REQUESTED** [review]
Thank you @nxp-shelley for resolving this.
But can you review these comments about the FlexSPI region sizes?

I did replicate the issue, and see this PR resolves it.  When building the ADC test below from the main branch, nothing prints out the console.  With this patch, the test passes and can be reflashed.
```
west build -b mimxrt1180_evk//cm33 tests/drivers/adc/adc_api/ --pristine
```

**Comment by hakehuang** [comment]
> @hakehuang , can you run our tests on this board, and ensure they all pass with this PR?

yes, tests are scheduled, will feedback once finish

File: soc/nxp/imxrt/imxrt118x/m33/mpu_regions.c (hunk 1 above)
Comment (nxp-shelley) [reply soc/nxp/imxrt/imxrt118x/m33/mpu_regions.c:10]: Yes, you are right.

File: soc/nxp/imxrt/imxrt118x/m33/mpu_regions.c (hunk 2 above)
Comment (nxp-shelley) [reply soc/nxp/imxrt/imxrt118x/m33/mpu_regions.c:11]: yes, you are right. but I think should be REGION_FLEXSPI2_SIZE ?

File: soc/nxp/imxrt/imxrt118x/m33/mpu_regions.c (hunk 3 above)
Comment (nxp-shelley) [reply soc/nxp/imxrt/imxrt118x/m33/mpu_regions.c:15]: yes, you are right

File: soc/nxp/imxrt/imxrt118x/m33/mpu_regions.c (hunk 2 above)
Comment (DerekSnell) [reply soc/nxp/imxrt/imxrt118x/m33/mpu_regions.c:11]: Agreed, thank you

**Review by DerekSnell — APPROVED** [review]
Looks great, @nxp-shelley .  Thank you for this fix!

**Review by danieldegrasse — APPROVED** [review]

**Review by mmahadevan108 — APPROVED** [review]

**Pull request merged**